docs/DeveloperGuide.md

```diff
@@ -1,11 +1,267 @@
+</p>
+
+If the user executes `search reservation; r/0; d/2020-03-13;`, the following result will be displayed in the image.
+
+<p align="center">
+    <img src="documentations\Sibing\SearchByIndexnDate.png">
+</p>
+
+The following sequence diagram shows the relevant interactions behind `search reservation`:
+
+<p align="center">
+    <img src="documentations\Sibing\SequenceDiagramforSearchReservation.png">
+</p>
+
+#### 1.4.2 Design Considerations
+##### Aspect: How search reservation executes
+* **Alternative 1 (current choice)**: Access reservation by `ReservationList#getReservation(reservationNumber)` if reservation number is provided,
+or do linear search along the reservation list for reservations with date matched.
+  + Pros: Easy to implement and understand.
+  + Cons: May have performance issues in terms of time usage when searching along the reservation list.
+
+* **Alternative 2**: Create a `HashMap` with date as the key and all Reservation objects in that date as the value.
+Once a new Reservation object is created, it will be added into the `HashMap` according to its date.
+  + Pros: Easier for searching in terms of date.
+  + Cons: Extra memory space is needed.
+
+
+##### Aspect: Data structure to support the search reservation feature.
+* **Alternative 1 (current choice)**: Display all all `Reservation` objects that contain the reservation number or date provided to the screen directly.
+  + Pros: Easy to implement and understand.
+  + Cons: The target objects are not really "contained" in a data structure.
+
+* **Alternative 2**: Use a `List` to store all `Reservation` objects that contain the reservation number or date provided.
+  + Pros: Objects are stored in a data structure, which is easier to do operation on that specific collections of objects.
+  + Cons: Extra memory space is needed.
+
+
+<a name="search-dish"></a>
+
+### 1.5 [Proposed] Search dish feature
+#### 1.5.1 Proposed implementation
+
+In restaurant daily report, a user can search from available menu items using the search dish commmand.
+
+The search dish command takes in a keyword denoted by k/, and searches all dish names for dishes containing the keyword.
+
+An example of the usage of search dish can be found below.
+
+Step 1. User launches application. An empty `Menu` is initialized.
+
+Step 2. User adds a menu item to the empty menu with `add dish; n/bacon pizza; i/cheese, bacon; p/7.00;`. Bacon pizza is now on the menu.
+
+Step 3. User adds more menu items to the menu. For example, they can `add dish; n/chicken rice; i/chicken, rice; p/4.00;` and `add dish; n/pasta with bacon; i/pasta, bacon; p/6.00;`
+
+Step 4. User searches the menu for any dish names containing a keyword. Let the user execute `search dish; k/bacon;`, the expected behavior can be found in the screenshot below.
+
+<p align="center">
+  <img src="https://user-images.githubusercontent.com/48315922/77872867-3e66e880-7216-11ea-9003-e04bda089f24.png">
+</p>
```

Comment:
![searchDish](https://user-images.githubusercontent.com/42376931/77989639-e33a1080-7351-11ea-8e07-bbfcef85acad.JPG)
Perhaps again these 3 commands do not have to be shown in the screenshot, simply explain that we have a list containing these items and show what search dish would return. I feel the details of how we add the dishes is not very relevant to this part of the DG as it seeks to explain how the search dish feature works.